### PR TITLE
ci: enable windows for testing heartbeat

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Enable testing for `heartbeat` on `windows`.

## Why is it important?

It was disabled since some tests were failing, and reported those tests but no further actions were done. 

Let's enable back so the @elastic/uptime can figure out what's going on

### CODE REVIEW

**Comment by andrewvc** [comment]
Apologies, I missed some build tags, mind if I push them directly here to make testing them easier?

**Comment by v1v** [comment]
Yes, please, contribute to this PR if needed 👍

**Comment by andrewvc** [comment]
I hope you don't mind @v1v , I've pushed directly to here to see if I can resolve the test failures with a few more build tags

**Review by andrewvc — APPROVED** [review]
LGTM

**Review by andrewvc — CHANGES_REQUESTED** [review]
Well, that was premature, I thought I'd seen them all passing

File: x-pack/heartbeat/scenarios/scenarios.go [hunk 1]
```diff
@@ -70,30 +69,5 @@ var Scenarios = &ScenarioDB{
 				}, func() {}, nil
 			},
 		},
-		{
-			Name: "simple-browser",
```
Comment (andrewvc) [inline x-pack/heartbeat/scenarios/scenarios.go:74]: Temporarily remove browser test from scenarios. This is a very new test that will come 
back in an improved form in https://github.com/elastic/beats/pull/30632

**Review by andrewvc — APPROVED** [review]
LGTM now

**Comment by v1v** [comment]
There are some unrequired commits from some other PRs, I'll try to fix them

**Comment by v1v** [comment]
I can see some errors in the logs:

```

[2022-08-31T21:06:34.702Z] >> python test: Unit Testing

[2022-08-31T21:06:59.250Z] ERROR: Could not install packages due to an EnvironmentError: [WinError 5] Access is denied: 'C:\\Users\\jenkins\\AppData\\Local\\Temp\\pip-uninstall-deorbl2h\\pip.exe'

[2022-08-31T21:06:59.250Z] Consider using the `--user` option or check the permissions.

[2022-08-31T21:06:59.250Z] 

[2022-08-31T21:07:40.233Z] warn: failed to upgrade pip (ignoring): running "null\build\ve\windows\Scripts\pip install -U pip" failed with exit code 1>> python test: Unit Testing - No tests found.
```

image

https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-32937/detail/PR-32937/6/pipeline/1446#step-2249-log-15


I don't know if those errors are genuine or flaky... and what's the reason the mage test does not fail, any hints?

**Comment by v1v** [comment]
@andrewvc , what's your feedback for https://github.com/elastic/beats/pull/32937#issuecomment-1233924710 ? It's the expected behaviour?

**Comment by andrewvc** [comment]
@v1v that's a bit of a mystery to me TBH, we don't do any python stuff outside of the standard tasks defined in libbeat's mage helpers.

I would have assumed that error would be present for all beats on windows if it's present for us. Are you certain other beats are running `mage pythonUnitTest` on windows?

Running locally on my win box (which I never normally do) I get a different, but similar error on metricbeat and heartbeat:

```
[notice] To update, run: C:\Users\AndrewCholakian\projects\beats\build\ve\windows\Scripts\python.exe -m pip install --upgrade pip
                                                                                         cd^C
C:\Users\AndrewCholakian\projects\beats\heartbeat [fix-remaining-win-issues +3 ~0 -0 !]> ^C
C:\Users\AndrewCholakian\projects\beats\heartbeat [fix-remaining-win-issues +3 ~0 -0 !]>
C:\Users\AndrewCholakian\projects\beats\heartbeat [fix-remaining-win-issues +3 ~0 -0 !]>
C:\Users\AndrewCholakian\projects\beats\heartbeat [fix-remaining-win-issues +3 ~0 -0 !]> cd ..\metricbeat
C:\Users\AndrewCholakian\projects\beats\metricbeat [fix-remaining-win-issues +3 ~0 -0 !]> mage pythonUnitTest
Generated fields.yml for metricbeat to C:\Users\AndrewCholakian\projects\beats\metricbeat\fields.yml
Generated fields.yml for metricbeat to C:\Users\AndrewCholakian\projects\beats\metricbeat\fields.yml
>> python test: Unit Testing
ERROR: To modify pip, please run the following command:
C:\Users\AndrewCholakian\projects\beats\build\ve\windows\Scripts\python.exe -m pip install -U pip

[notice] A new release of pip available: 22.2.1 -> 22.2.2
[notice] To update, run: C:\Users\AndrewCholakian\projects\beats\build\ve\windows\Scripts\python.exe -m pip install --upgrade pip
warn: failed to upgrade pip (ignoring): running "C:\Users\AndrewCholakian\projects\beats\build\ve\windows\Scripts\pip install -U pip" failed with exit code 1
[notice] A new release of pip available: 22.2.1 -> 22.2.2
[notice] To update, run: C:\Users\AndrewCholakian\projects\beats\build\ve\windows\Scripts\python.exe -m pip install --upgrade pip
ERROR: Could not find a version that satisfies the requirement pywin32==227; sys_platform == "win32" (from docker) (from versions: 302, 303, 304)
ERROR: No matching distribution found for pywin32==227; sys_platform == "win32"

[notice] A new release of pip available: 22.2.1 -> 22.2.2
[notice] To update, run: C:\Users\AndrewCholakian\projects\beats\build\ve\windows\Scripts\python.exe -m pip install --upgrade pip
Error: running "C:\Users\AndrewCholakian\projects\beats\build\ve\windows\Scripts\pip install --quiet -Ur C:\Users\AndrewCholakian\projects\beats/libbeat/tests/system/requirements.txt" failed with exit code 1
```

They're both running pip commands.

TBH, I would like to keep those disabled on windows, the value those tests add is minimal, not nearly high enough to justify the maintenance cost, esp. given that we've never had a windows specific bug in many years.

**Comment by v1v** [comment]
>  Are you certain other beats are running mage pythonUnitTest on windows?

Yes, with similar outcome but running some tests afterwards, for instance, https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-32937/detail/PR-32937/6/pipeline/1446#step-2249-log-15 for `metricbeat` 


I assume, since it happens for other components it's okey to merge this PR. 

Do you want me to do it now?

**Comment by andrewvc** [comment]
Yes, ++ on merging

**Pull request merged**